docs/design/proposals/provisioned-service/provisioned-service.md

```diff
@@ -0,0 +1,108 @@
+# Service Binding Specification for Kubernetes
+
+## Problem Statement
+
+Application developers require specific values to connect their
+applications to various services. In order to facilitate this connection
+problem, the [Service Binding Specification for
+Kubernetes](https://servicebinding.io) defines Provisioned
+Services as a standard way to expose connection information from backing
+services. Supporting Provisioned Services within AWS Controllers for
+Kubernetes makes application connectivity to ACK-backed services easy
+and seamless.
+
+## Existing Solutions
+
+AWS Controllers for Kubernetes have support for a [FieldExport
+API](https://github.com/aws-controllers-k8s/community/blob/main/docs/design/proposals/native-binding/native-binding.md).
+This can be used to create a Secret that can be used for [Direct Secret
+Reference as per the
+specification](https://github.com/servicebinding/spec#direct-secret-reference).
+This solution needs more effort from the application developers.
+
+The FieldExports resources required for application developers are the
+same, but every developer needs to repeat this same configuration for
+every application they deploy. The proposed solution avoids this
+repetition through Provisioned Services.
+
+## Proposed Solution
+
+The proposed solution is to make various ACK resources become
+Provisioned Services. For example, make the DBInstance resource provided
+by the RDS controller a Provisioned Service. Each controller would use
+their main custom resource as a Provisioned Service resource. For each
+Provisioned Service, identify the fields that need to be exported to
+create the Secret resource.
+
+The application developer experience is explained in this demo video:
+[https://www.youtube.com/watch?v=AXXWv7N12JM](https://www.youtube.com/watch?v=AXXWv7N12JM)
+
+(This is created as part of a [Proof of
+Concept](https://github.com/aws-controllers-k8s/community/issues/1289))
+
+The structure change for DBInstance:
+
+```
+// ServiceBindingSecretReference defines a mirror of corev1.LocalObjectReference
+type ServiceBindingSecretReference struct {
+	// Name of the referent secret.
+	// More info: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names
+	Name string `json:"name"`
+}
+
+// DBInstanceStatus defines the observed state of DBInstance
+type DBInstanceStatus struct {
+	// Binding exposes the Secret for Service Binding to conform the Provisioned Service
+	// as per the Service Binding Specification for Kubernetes.
+	// Ref. https://github.com/servicebinding/spec#provisioned-service
+	Binding *ServiceBindingSecretReference `json:"binding,omitempty"`
```

Comment:
We would want to place this field within the `Status.ACKResourceMetadata` subfield. That way, it won't conflict with any resources that might have a `Binding` field already in their status...

Reply:
The Service Binding Specification expects the Provisioned Service (e.g., `DBInstance`) to point to the Secret resource from `.status.binding.name` attribute. I think we can confirm if that field is not conflicting with any existing fields.

Reply:
@baijum the problem is that the `Status` and `Spec` fields for all ACK resources are automatically generated by looking at the AWS service API's model definition. We don't know if there are APIs that have a `Binding` field in any Create operation's Output shape. If there were, that would conflict with this.

Reply:
Can we check if the`.status.binding.name` attribute exists for any API? Also is it possible to reserve this field for service binding, so that it will not conflict in the future?

Reply:
We could do this, sure, but the problem is we cannot predict the future. Since ACK CRDs are generated from the upstream API models, what happens if an upstream service API adds a Binding field in the future? This is why we place these kinds of fields into `Status.ACKResourceMetadata` to ensure that we don't conflict.

Reply:
I have created a proposal to the spec here: https://github.com/servicebinding/spec/pull/223
Let me know if creating a Secret resource with the same name as that of `DBInstance` resource (or any other service) would be acceptable.

Reply:
@baijum I left a comment in the other PR. I would be OK with a separate reconciler (ala the `FieldExportReconciler` we already have in ACK) that would create/manage Secret resources like servicebinding.io expects.

Reply:
+1 with Jay's comment here. This functionality seems similar enough to `FieldExport`, in that it accesses CRD fields and outputs into a `Secret`, that I believe it could simply be handled by that reconciler. In fact, we could add this feature by simply adding an optional field within the `FieldExport` CRD that provides an alternative to `from.path`, instead being `from.bindingSpecification: true/false` (name TBD). This would sidestep the need to reserve the `.status.binding` field.